app/routes/task_routes.py

```diff
@@ -1 +1,91 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint,abort, make_response, request, Response
+from app.models.task import Task
+from app.routes.helpers import validate_model, create_new_model_dict, send_slack_message
+from ..db import db
+
+
+bp = Blueprint("tasks_bp", __name__, url_prefix="/tasks")
+
+@bp.post("")
+def create_task():
+    request_body = request.get_json()
+
+    return create_new_model_dict(Task,request_body)
+
+@bp.get("")
+def get_all_task():
+    query = db.select(Task)
+
+    title_param = request.args.get("title")
+    if title_param:
+        query = query.where(Task.title.ilike(f"%{title_param}%"))
+
+    description_param = request.args.get("description")
+    if title_param:
+        query = query.where(Task.description.ilike(f"%{description_param}%"))
+
+    is_complete_param = request.args.get("is_complete")
+    if title_param:
+        query = query.where(Task.completed_at.ilike(f"%{is_complete_param}%"))
+
+    sort_param = request.args.get('sort')
+    if sort_param == 'asc':
+        query = query.order_by(Task.title.asc())
+    elif sort_param == 'desc':
+        query = query.order_by(Task.title.desc())
+
+    tasks = db.session.scalars(query.order_by(Task.id))
+
+    tasks_response = [task.to_dict() for task in tasks]
```

Comment:
Nice work on handling the filtering logic for this route and the goal route. To D.R.Y. this up some more we could use similar logic to the`get_models_with_filters` helper function that we implemented for flasky.